src/pages/index.js

```diff
@@ -76,21 +90,113 @@ const popupWithImage = new PopupWithImage("#image-popup");
 popupWithImage.setEventListeners();
 
 //Section
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      renderCard(cardData);
-    },
-  },
-  "cards__list"
-);
+let section;
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardEl = renderCard(cardData);
+          section.addItem(cardEl);
+        },
+      },
+      "cards__list"
+    );
+    section.renderItems();
+  })
+
+  .catch((error) => {
+    console.error(error);
+  });
+
+api
+.getUserInfo()
+  .then((user) => {
+    userInfo.setUserInfo(user.name, user.about);
+    userInfo.setAvatar(user.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-section.renderItems();
 //UserInfo
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.description);
-  editPopup.close();
+  editPopup.setLoading(true);
+  api.editUserInfo(inputValues).then(() => {
+    userInfo.setUserInfo(inputValues.name, inputValues.description);
+    editPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    editPopup.setLoading(false);
+  });
+}
+
+//EditAvatar
+
+const updateAvatarForm = new PopupWithForm("#profile-image-modal", handleAvatarSubmit);
+function handleAvatarSubmit(inputValues) {
+  updateAvatarForm.setLoading(true);
+  api.updateAvatar(inputValues.link).then((user) => {
+    userInfo.setAvatar(user.avatar);
+    updateAvatarForm.setLoading(false);
+  });
+  updateAvatarForm.close();
+}
+
+const editPencilIcon = document.querySelector("#avatar-edit-button");
+
+editPencilIcon.addEventListener("click", () => {
+  updateAvatarForm.open();
+});
+
+updateAvatarForm.setEventListeners();
+//PopupWithConfirmation
+
+const deleteCardPopup = new PopupWithConfirmation("#delete-card-modal");
+
+function handleDeleteButton(cardID, card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors